contents/dotfiles/Dotfiles.md

```diff
@@ -0,0 +1,232 @@
+
+##### Feature 3.1: Prompt
+
+The `PS1` environment variable provides information for configuring and styling the Shell prompt. 
+By default, `PS1` is set to display only the username, host name and current working directory in the prompt.
+
+<pic src="default-prompt.png" alt="Default prompt" width="55%">
+
+  <sub> _Figure 2. Default prompt_ </sub>
+
+</pic>
+
+`PS1` can be [configured](https://www.cyberciti.biz/tips/howto-linux-unix-bash-shell-setup-prompt.html) to add (or remove) features, such as status code of the previous command, git branch, git status, dates, etc. In Figure 3 below, the prompt is able to detect and show the git branch and status when it is inside a git repository, a useful and convenient feature for developers.
+
+<pic src="enhanced-prompt.png" alt="Enhanced prompt" width="55%">
+
+  <sub> _Figure 3. Enhanced prompt_ </sub>
+
+</pic>
+
+##### Feature 3.2: Tab Completion
+
+Tab completion is a feature that allows the user to select options from a drop down menu. This is a default feature when using `zsh`, but can be configured using dotfiles when using other shells like `bash`.
+
+<pic src="cd-autocomplete.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 4. `cd` tab completion [(Tutorial)](https://scriptingosx.com/2019/07/moving-to-zsh-part-5-completions/)_ </sub>
+
+</pic>
+
+Tab completion can also be used for other purposes, such as selecting flags from a drop down menu, complete with an accompanying description.
+
+<pic src="git-flag-autocomplete.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 5. Flag autocomplete_ </sub>
+
+</pic>
+
+##### Feature 3.3: Syntax Highlighting
+
+Dotfiles are also often used to style and colorise the terminal. This is not just for aesthetic reasons - the use of color enhances readability of programs and makes it easier to debug. One such example would be the use of **syntax highlighting** as seen in Figure 6 below. It is especially useful for long shell commands, escape sequences or string interpolation.
+
+<pic src="syntax-highlighting.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 6. Syntax Highlighting [(Tutorial)](https://github.com/zsh-users/zsh-syntax-highlighting)_ </sub>
+
+</pic>
+
+### Getting Started
+
+While dotfiles are easy to setup and configure, they can become disorganised over time as a developer as more and more configurations are added. As much as possible, they should be managed like any other software engineering project, applying software design principles such as modularisation to make them extensible and future-proof.
```

Comment:
Can standardise all dotfiles -> Dotfiles 
or 
standardise to use dotfiles in the middle of sentences instead of Dotfiles (as there are some sections where u use Dotfiles in mid sentence)